datalad/core/distributed/clone.py

```diff
@@ -1206,20 +1207,32 @@ def _get_installationpath_from_url(url):
     from a URL, analog to what `git clone` does.
     """
     ri = RI(url)
-    if isinstance(ri, (URL, DataLadRI)):  # decode only if URL
-        path = ri.path.rstrip('/')
-        path = urlunquote(path) if path else ri.hostname
-        if '/' in path:
-            path = path.split('/')
-            if path[-1] == '.git':
-                path = path[-2]
-            else:
-                path = path[-1]
+
+    def get_last_path(url):
+        "Return the latest part of the Path(url) which is not .git"
+        for part in Path(url).parts[::-1]:
+            if part != '.git':
+                return part
+        return ''
+
+    # consider path from RI, if degenerate - hostname, and if that
+    # one is not known -- URL itself as if it was a Path, but only
+    # the last (non .git) component
+    for cand in (
+        urlunquote(getattr(ri, 'path', '')),
```

Comment:
well, thinking about it -- may be `urlunquote` should be URL/DataLadRI specific? but I wonder - don't we decode it there?    worth adding an explicit test and ensure that we do not alter precious /home/%e0 etc while decoding them when in URLs... may be later ;)